lib/path.js

```diff
@@ -176,23 +211,21 @@ win32.normalize = function(path) {
 
 win32.isAbsolute = function(path) {
   assertPath(path);
-
-  var result = splitDeviceRe.exec(path),
-      device = result[1] || '',
-      isUnc = !!device && device.charAt(1) !== ':';
-  // UNC paths are always absolute
-  return !!result[2] || isUnc;
+  return win32StatPath(path).isAbsolute;
 };
 
 win32.join = function() {
-  function f(p) {
-    if (typeof p !== 'string') {
+  var paths = [];
+  for (var i = 0; i < arguments.length; i++) {
+    var arg = arguments[i];
+    if (typeof arg !== 'string') {
```

Comment:
I think `assertPath` would be the better choice since that will also add the type of the bad argument to the error message.